### PR TITLE
Add test to check if syslog message are parsed correctly in the `alerts.json` file

### PR DESCRIPTION
**Pull request by fedepacher**
|Related issue|
|-------------|
|      https://github.com/wazuh/wazuh-qa/issues/3095      |

## Description

This PR aims to add a new test to check that the `syslog` messages are parsed correctly in the `alerts.json` file. 

### Added

-  test_syslog_message_parser.py
- configuration_syslog_message_parser.yaml
- cases_syslog_message_parser.yaml

### Updated

- syslog_simulator.py
- run_simulaor.py

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @fedepacher (Developer)  |           | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33935/console) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33938/console) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33939/console)  | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/10024908/r1-3095.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/10024914/r2-3095.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/10024917/r3-3095.zip)  |         |         | Nothing to highlight |
| @Rebits  (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+# Syslog message IPV6 values test configurations (t1)
+t1_configurations_parameters, t1_configurations_metadata, t1_cases_ids = get_test_cases_data(t1_cases_path)
+t1_configurations = load_configuration_template(t1_configurations_path, t1_configurations_parameters,
+                                                t1_configurations_metadata)
+
+
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configurations_metadata),
+                         ids=t1_cases_ids)
+def test_syslog_message_ipv6(configuration, metadata, set_wazuh_configuration, truncate_event_logs,
```
Comment (jmv74211) [inline tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py:79]: You are testing ipv4 and ipv6 in test cases, so the name of this test is not valid

File: tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    '''
+    description: Check if 'wazuh-remoted' can receive syslog messages through the socket.
+
+    test_phases:
+        - setup:
+            - Apply ossec.conf configuration changes according to the configuration template and use case.
+            - Truncate wazuh event logs.
+            - Restart wazuh-manager service to apply configuration changes.
+        - test:
+            - Check that the messages are parsed correctly in the alerts.json file.
```
Comment (jmv74211) [inline tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py:90]: You are checking archives file not alerts one

File: tests/integration/test_remoted/test_socket_communication/data/test_cases/cases_syslog_message_parser.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+- name: Syslog_message_ipv6_test
```
Comment (jmv74211) [inline tests/integration/test_remoted/test_socket_communication/data/test_cases/cases_syslog_message_parser.yaml:1]:
```suggestion
- name: syslog_message_ipv6
```

File: tests/integration/test_remoted/test_socket_communication/data/test_cases/cases_syslog_message_parser.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    PROTOCOL: tcp
+  metadata:
+    # syslog simulator parameters
+    address: 0000:0000:0000:0000:0000:0000:0000:0001
+    message: IPv6 test msg
+    port: 5050
+    protocol: tcp
+    messages_number: 1
+
+- name: Syslog_message_ipv4_test
```
Comment (jmv74211) [inline tests/integration/test_remoted/test_socket_communication/data/test_cases/cases_syslog_message_parser.yaml:14]:
```suggestion
- name: syslog_message_ipv4
```

File: tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    # Read the events log data
+    events_data = file.read_file(ARCHIVES_JSON_PATH).split('\n')
+
+    message = metadata['message']
+    location = metadata['address']
+    find_msg = (fr'"full_log":"{message}".*"location":"{location}"').replace('"', r'\"')
+
+    event_msg = [event for event in events_data if bool(re.match(fr".*{find_msg}.*", event))]
+
+    assert len(event_msg) == metadata['messages_number'], f"The events were not parsed as: {find_msg}"
```
Comment (jmv74211) [inline tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py:147]:
```suggestion
    assert len(event_msg) == metadata['messages_number'], 'The event's format is not the expected one'
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py (hunk 5 above)
Comment (fedepacher) [reply tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3609/commits/b6c0aac937531f132aeb56fd1e5f4d49f589fd39)

File: tests/integration/test_remoted/test_socket_communication/data/test_cases/cases_syslog_message_parser.yaml (hunk 3 above)
Comment (fedepacher) [reply tests/integration/test_remoted/test_socket_communication/data/test_cases/cases_syslog_message_parser.yaml]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3609/commits/b6c0aac937531f132aeb56fd1e5f4d49f589fd39)

File: tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py (hunk 2 above)
Comment (fedepacher) [reply tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3609/commits/b6c0aac937531f132aeb56fd1e5f4d49f589fd39)

File: tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py (hunk 1 above)
Comment (fedepacher) [reply tests/integration/test_remoted/test_socket_communication/test_syslog_message_parser.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3609/commits/b6c0aac937531f132aeb56fd1e5f4d49f589fd39)

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**